task_manager/forms.py

```diff
@@ -1,23 +1,63 @@
+from datetime import datetime
+
 from flask_wtf import FlaskForm
-from wtforms import StringField, PasswordField, BooleanField, SubmitField
-from wtforms.validators import DataRequired, Email, EqualTo, ValidationError
+from wtforms import StringField, PasswordField, BooleanField, SubmitField, \
+    RadioField, DateField, TimeField
+from wtforms.validators import DataRequired, Email, EqualTo, Length, \
+    ValidationError, InputRequired
 
 from task_manager.models.user import User
 
 
+class CreateTaskForm(FlaskForm):
+    title = StringField('Title', validators=[DataRequired(),
+                                             Length(min=4, max=25)])
+    priority = RadioField('Priority', choices=['1', '2', '3', '4', '5'],
+                          default='1', validators=[InputRequired()],
+                          coerce=int)
+    description = StringField('Description', validators=[DataRequired()])
+    start_date = DateField('Start Date', validators=[DataRequired()])
+    start_time = TimeField('Start Time', validators=[DataRequired()])
+    end_date = DateField('End Date', validators=[DataRequired()])
+    end_time = TimeField('End Time', validators=[DataRequired()])
+
+    @property
+    def start(self):
+        if self.start_date.data and self.start_time.data:
+            return datetime.combine(self.start_date.data, self.start_time.data)
+
+    @property
+    def end(self):
+        if self.end_date.data and self.end_time.data:
+            return datetime.combine(self.end_date.data, self.end_time.data)
+
+    # def validate_end_date(self, field):
```

Comment:
how does this validate the endtime > starttime?

Reply:
i commented this out since it was interfering with a testcase. i didn't have enough time to debug and had to get our coverage up from 50%.